### PR TITLE
[Cleanup] Reorganize main reconciliation context

### PR DESCRIPTION
**Pull request by ajanikow**

### CODE REVIEW

File: pkg/deployment/server_api.go [hunk 1]
```diff
@@ -44,8 +44,8 @@ func (d *Deployment) Namespace() string {
 	return d.apiObject.Namespace
 }
 
-// Mode returns the mode of the deployment.
-func (d *Deployment) Mode() api.DeploymentMode {
+// GetAgencyClientsWithPredicate returns the mode of the deployment.
```
Comment (informalict) [inline pkg/deployment/server_api.go:47]: GetMode instead GetAgencyClientsWithPredicate

**Review by informalict — APPROVED** [review]
LGTM with small comment

File: pkg/deployment/server_api.go (hunk 1 above)
Comment (nikita-vanyasin) [inline pkg/deployment/server_api.go:47]: I guess there should be GetMode instead

**Review by nikita-vanyasin — CHANGES_REQUESTED** [review]
All LGTM except small comment fix for `GetMode()` function.

These new TODO comments in tests at [235fc6f789315bea8d9d18981b399a7c880bac8cc38421cfba0a91a818e96656R97](https://github.com/arangodb/kube-arangodb/pull/901/files#diff-235fc6f789315bea8d9d18981b399a7c880bac8cc38421cfba0a91a818e96656R98)
  also does look a bit confusing but I guess this is some old code moved from one place to another?

**Review by nikita-vanyasin — APPROVED** [review]

**Comment by ajanikow** [comment]
> All LGTM except small comment fix for `GetMode()` function.
> 
> These new TODO comments in tests at [235fc6f789315bea8d9d18981b399a7c880bac8cc38421cfba0a91a818e96656R97](https://github.com/arangodb/kube-arangodb/pull/901/files#diff-235fc6f789315bea8d9d18981b399a7c880bac8cc38421cfba0a91a818e96656R98) also does look a bit confusing but I guess this is some old code moved from one place to another?

We are not implementing not-used test functions - it is desired to do panic with them to make sure that they are not used.

**Pull request merged**